unit_scaling/functional.py

```diff
@@ -201,13 +201,27 @@ def layer_norm(
     unsupported_args=["alpha"],
 )
 def add(
-    input: Tensor, other: Tensor, alpha: int = 1, out: Optional[Tensor] = None
+    input: Tensor,
+    other: Tensor,
+    constraint: Optional[TernaryConstraint] = gmean,
+    alpha: int = 1,
+    out: Optional[Tensor] = None,
 ) -> Tensor:
     input_broadcast_size, other_broadcast_size = _get_broadcast_sizes(input, other)
-    input = scale_bwd(input, input_broadcast_size**-0.5)
-    other = scale_bwd(other, other_broadcast_size**-0.5)
+    grad_input_scale = input_broadcast_size**-0.5
+    grad_other_scale = other_broadcast_size**-0.5
+    fwd_scale = 2**-0.5
+    if constraint:
```

Comment:
I have to use this slightly nasty construct (used elsewhere) to handle the different kinds of constraints one is allowed to pass in

Reply:
Ah, yes, I remember wondering if this boilerplate could be refactored into an `apply_constraint(constraint, ...)` somehow.